docs/arch/monitor.md

```diff
@@ -0,0 +1,47 @@
+# Monitor
+
+## Contributors
+
+- Sarah THEOULLE
+- Pauline CONTAT
+- Thomas BROINE
+- Baptiste BRONSIN
+
+## Functionnalities
+
+- Listening events on a Git remote repository
+- Recognition of the event and adapt the action according to the type of the event
+- Calling the controller through an external API
+
+## What
+
+A monitor must listen to some events from a git remote repository and do some actions onto it.
+We need to recognise the type of event, a `PR` (Pull Request) at first.
+Depending on the type of the event, a yaml data will be generated and sent to the controller.
+
+## Why
+
+The goal is to trigger the controller to launch a CI accordingly to the detected event from the distant repository.
+The `yaml` format has been chosen because the controller requires this type of file.
+
+## How
+
+![schema of the monitor gobal architecture](image-1.png)
+
+**Set Up the Git Repository:** Configure the Git remote repository to detect specific events (like Pull Requests).
+https://github.com/XAMPPRocky/octocrab
+
+https://octocat-rs.github.io/book/ 
+
+**Develop the Event Listener:** Create a api on a server that listens for incoming webhook notifications from the Git repository. This server will handle incoming events and process them.
+
+**Recognize and Handle Events:** Implement logic to recognize different types of events (starting with Pull Requests) and take appropriate actions based on the event type.
+
+**Generate YAML Data:** Based on the recognized event, generate the required YAML data.
+
+**Send Data to the Controller:** Use an API to send the generated YAML data to the controller.
```

Comment:
Can you detail how you are going to fetch our API. Please refer to our Open API v3 spec here : https://github.com/dev-sys-do/sealci/pull/7. Really important, let us know if you need other endpoints/to modify current endpoints

Reply:
I developed the API route used and the body data.